### PR TITLE
[PD-2100] user management - dynamically update based on changes to app_access

### PR DESCRIPTION
**Pull request by lemniscate613**
# Summary

The User Management front end will adjust as we add new apps.

__1. List on Activities view is dynamic__

screen shot 2016-02-04 at 10 56 53 am

__2. Accordions on Edit/Add Role view are dynamic__

screen shot 2016-02-04 at 10 57 03 am

# Other
* Refactored API endpoints that are relevant to this PR. The data these return is now simpler:
  * /manage-users/api/activities/
  * /manage-users/api/roles/
* Updated related Python and JS tests.
* Refactored activities list with `groupBy()` -- thanks to @tdphillips for the suggestion.
* Reorganized files according to @darrint's plan (i.e. phase out `gulp/src/util` folder)
* Eliminated ~55 queries in APIs by using `select_related()` and `prefetch_related()` -- thanks to @JLMcLaugh for the tip.
* Fixed some image references in styles so we don't get mixed content warnings (i.e. trying to access HTTP over HTTPS).

# Review

Python tests:
```
dkm test myjobs.tests.test_manage_users.ManageUsersTests
```

JS tests:
```
dkgg build
```

Create temporary app_access with related activities:

```
dkm shell
```

Then:
```
from myjobs.models import AppAccess
from seo.models import Company
from myjobs.models import Activity

app_access = AppAccess.objects.create(name="Example")
company = Company.objects.filter(name="DirectEmployers Association").first()
company.app_access.add(app_access)

activities = Activity.objects.bulk_create([
   Activity(name="example activity 1", app_access=app_access),
   Activity(name="example activity 2", app_access=app_access)])
```

See new app_access with related activities in the Activities view and in the Edit/Add Role view.

Delete temporary app_access:

```
app_access = AppAccess.objects.filter(name="Example")
app_access.delete()
```

Refresh to see changes.

# Yes

tabletop_roleplaying

### CODE REVIEW

**Comment by aspidites** [comment]
Still missing JavaScript tests (specs).

File: myjobs/views.py [hunk 1]
```diff
@@ -632,62 +649,96 @@ def api_get_roles(request):
...
+                            .filter(app_access__in = app_access_for_company))
+
+    # Retrieve users that can be assigned to these roles. In other words,
+    # users already assigned to roles associated with this company
+    available_users_as_queries = []
+    for role in roles:
+        role_id_temp = role.id
+        users = User.objects.filter(roles__id=role_id_temp).distinct()
+        for user in users:
+            # Make sure available_users is distinct
```
Comment (tdphillips) [inline myjobs/views.py]: You're already telling the ORM to return distinct users.

File: myjobs/views.py [hunk 2]
```diff
@@ -632,62 +649,96 @@ def api_get_roles(request):
...
+    available_activities = (Activity
+                            .objects
+                            .filter(app_access__in = app_access_for_company))
+
+    # Retrieve users that can be assigned to these roles. In other words,
+    # users already assigned to roles associated with this company
+    available_users_as_queries = []
+    for role in roles:
+        role_id_temp = role.id
+        users = User.objects.filter(roles__id=role_id_temp).distinct()
```
Comment (tdphillips) [inline myjobs/views.py]: Instead of iterating over all roles, this query could probably be filtering for `roles__in=roles`

File: myjobs/views.py [hunk 3]
```diff
@@ -632,62 +649,96 @@ def api_get_roles(request):
...
+        for user in users:
+            # Make sure available_users is distinct
+            if user not in available_users_as_queries:
+                available_users_as_queries.append(user)
+    # available_users should be distinct list of user queries
+    available_users = []
+    for user in available_users_as_queries:
```
Comment (tdphillips) [inline myjobs/views.py]: Given the previous two comments, we could do away with `available_users_as_queries` and construct `available_users` directly.

Comment (JLMcLaugh) [inline myjobs/views.py]: :+1:

**Comment by lemniscate613** [comment]
Thanks @aspidites Just added / renamed some tests: https://github.com/DirectEmployers/MyJobs/commit/8cc1cc84f8e05ab1373b25b058e5ab76f7e77060

File: myjobs/views.py (hunk 2 above)
Comment (lemniscate613) [inline myjobs/views.py]: Excellent, excellent point. I just updated it: https://github.com/DirectEmployers/MyJobs/commit/4edc36fc085b2945c9f21d950acb09c2cc4dc6f8

**Comment by tdphillips** [comment]
Merge conflicts.

File: myjobs/views.py (hunk 3 above)
Comment (lemniscate613) [inline myjobs/views.py]: Awesome idea @tdphillips: https://github.com/DirectEmployers/MyJobs/commit/4edc36fc085b2945c9f21d950acb09c2cc4dc6f8

**Pull request merged**